ONSdigital/census-rh-ui
# Cr 1826 fix name field spaces

# Motivation and Context
Adds trapping for spaces only and removal of leading/trailing spaces in non-regex checked fields

# What has changed
Update to response validation for first name, last name in all 'Enter name' pages (all postal fulfilment journeys)
Update to response validation for town name in transient journey
Update to response validation for 'add room number' in CE fulfilment journeys (note room number field can be empty, but should disregard just spaces)
Update to response validation for web form 'name' field
Update to response validation for web chat 'screen name' field
New and updated unit tests for above

No Cucumber updates required

Other text fields in the authentication and fulfilment journeys do not require an update as they are covered by their existing regexes

# How to test?
Try entering just spaces in the fields - check that response is treated as if the field is empty

# Links
https://collaborate2.ons.gov.uk/jira/browse/CR-1826
